PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,438 @@
+                    i, n.val, n.color)
+                n.color = self.canvas.itemconfigure(n.display_shape, 'fill')
+
+        self.window.update()
+
+    def randomFill(self):
+        callEnviron = self.createCallEnvironment()        
+        # Clear the list so new values can be entered
+        self.list=[] 
+        size = self.size
+        
+        # Create a list of random numbers and sort them
+        a = [random.randrange(90) for i in range(size)]
+        a.sort()
+        
+        # Append and draw them to the list and draw them
+        for i in a:
+            self.list.append(drawable(i))
+        
+        self.display()         
+        self.cleanUp(callEnviron)
+        
+    def newArraySize(self, val):
+        callEnviron = self.createCallEnvironment()                
+        # Clear Array and reset size and list
+        self.canvas.delete("all")
+        self.size = val
+        self.list = []        
+        
+        for i in range(val):  # Draw new grid of cells
+            self.createArrayCell(i) 
+        
+        self.window.update()
+        self.cleanUp(callEnviron)
+
+    def find(self, val):
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
+        lo = 0                             #Point to lo
+        indexLo = self.createIndex(lo, 'lo', level=1)
+        callEnviron |= set(indexLo)
+        hi = len(self.list)-1              # Point to hi
+        indexHi = self.createIndex(hi, 'hi', level = 3)
+        callEnviron |= set(indexHi)
+        mid = (lo + hi) // 2               # Point to the midpoint
+        indexMid = self.createIndex(mid, 'mid', level=2)
+        callEnviron |= set(indexMid)
+        while lo <= hi:
+            mid = (lo + hi) // 2           # Select the midpoint
+            if self.list[mid].val == val:  # Did we find it at midpoint?
+                self.stopAnimations()
+                self.window.update()
+                return mid                 # Return the value found
+
+            elif self.list[mid].val < val: # Is item in upper half?
+                deltaXLo = (mid - lo) + 1
+                self.moveItemsBy(indexLo, (self.CELL_SIZE*deltaXLo, 0))
+                lo = mid + 1               # Yes, raise the lo boundary
+                deltaXMid = ((hi - lo) // 2) + 1
+                self.moveItemsBy(indexMid, (self.CELL_SIZE*deltaXMid, 0))
+
+            else:                         # Is item in lower half?
+                deltaXHi = (mid-hi) - 1
+                self.moveItemsBy(indexHi, (self.CELL_SIZE*deltaXHi, 0))
+                hi = mid - 1              # Yes, lower the hi boundary
+                deltaXMid = ((lo-hi)//2)-1
+                self.moveItemsBy(indexMid, (self.CELL_SIZE*deltaXMid, 0))
+
+        self.stopAnimations()
+        self.window.update()
+        self.cleanUp(callEnviron)
+        return lo                         # val not found
```

Comment:
While the data in the PriorityQueue is sorted and binary search could be used to find a particular item, the programming interface usually does not provide that operation.  The `find()` method is not needed for this data structure.